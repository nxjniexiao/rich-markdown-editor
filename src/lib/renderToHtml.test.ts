import renderToHtml from "./renderToHtml";

test("renders an empty string", () => {
  expect(renderToHtml("")).toBe("");
});

test("renders plain text as paragraph", () => {
  expect(renderToHtml("plain text")).toMatchSnapshot();
});

test("renders blockquote", () => {
  expect(renderToHtml("> blockquote")).toMatchSnapshot();
});

test("renders code block", () => {
  expect(
    renderToHtml(`
    this is indented code
`)
  ).toMatchSnapshot();
});

test("renders code fence", () => {
  expect(
    renderToHtml(`\`\`\`javascript
this is code
\`\`\``)
  ).toMatchSnapshot();
});

test("renders checkbox list", () => {
  expect(
    renderToHtml(`- [ ] unchecked
- [x] checked`)
  ).toMatchSnapshot();
});

test("renders bullet list", () => {
  expect(
    renderToHtml(`- item one
- item two
  - nested item`)
  ).toMatchSnapshot();
});

test("renders info notice", () => {
  expect(
    renderToHtml(`:::info
content of notice
:::`)
  ).toMatchSnapshot();
});

test("renders warning notice", () => {
  expect(
    renderToHtml(`:::warning
content of notice
:::`)
  ).toMatchSnapshot();
});

test("renders tip notice", () => {
  expect(
    renderToHtml(`:::tip
content of notice
:::`)
  ).toMatchSnapshot();
});

test("renders headings", () => {
  expect(
    renderToHtml(`# Heading 1

## Heading 2

### Heading 3

#### Heading 4`)
  ).toMatchSnapshot();
});

test("renders horizontal rule", () => {
  expect(renderToHtml(`---`)).toMatchSnapshot();
});

test("renders image", () => {
  expect(
    renderToHtml(`![caption](https://lorempixel.com/200/200)`)
  ).toMatchSnapshot();
});

test("renders image with alignment", () => {
  expect(
    renderToHtml(`![caption](https://lorempixel.com/200/200 "left-40")`)
  ).toMatchSnapshot();
});

test("renders table", () => {
  expect(
    renderToHtml(`
| heading | centered | right aligned |
|---------|:--------:|--------------:|
|         | center   |               |
|         |          |      bottom r |
`)
  ).toMatchSnapshot();
});

test("renders bold marks", () => {
  expect(renderToHtml(`this is **bold** text`)).toMatchSnapshot();
});

test("renders code marks", () => {
  expect(renderToHtml(`this is \`inline code\` text`)).toMatchSnapshot();
});

test("renders highlight marks", () => {
  expect(renderToHtml(`this is ==highlighted== text`)).toMatchSnapshot();
});

test("renders italic marks", () => {
  expect(renderToHtml(`this is *italic* text`)).toMatchSnapshot();
  expect(renderToHtml(`this is _also italic_ text`)).toMatchSnapshot();
});

test("renders template placeholder marks", () => {
  expect(renderToHtml(`this is !!a placeholder!!`)).toMatchSnapshot();
});

test("renders underline marks", () => {
  expect(renderToHtml(`this is __underlined__ text`)).toMatchSnapshot();
});

test("renders link marks", () => {
  expect(
    renderToHtml(`this is [linked](https://www.example.com) text`)
  ).toMatchSnapshot();
});

test("renders underline marks", () => {
  expect(renderToHtml(`this is ~~strikethrough~~ text`)).toMatchSnapshot();
});

test("renders ordered list", () => {
  expect(
    renderToHtml(`1. item one
1. item two`)
  ).toMatchSnapshot();

  expect(
    renderToHtml(`1. item one
2. item two`)
  ).toMatchSnapshot();
});

test("renders color", () => {
  expect(
    renderToHtml(
      "<<rgb(240, 107, 5) <<<rgba(163, 67, 31, 0.2) 测试>>>>><<rgb(136, 49, 204) <<<rgba(74, 82, 199, 0.2) 文字>>>>><<rgb(233, 30, 44) <<<rgba(200, 21, 182, 0.2) 背景>>>>><<rgb(3, 135, 102) <<<rgba(3, 135, 102, 0.2) 颜色>>>>>"
    )
  ).toMatchSnapshot();
});
