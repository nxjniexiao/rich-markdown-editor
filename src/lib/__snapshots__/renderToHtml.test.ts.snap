// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders blockquote 1`] = `
"<blockquote>
<p>blockquote</p>
</blockquote>"
`;

exports[`renders bold marks 1`] = `"<p>this is <strong>bold</strong> text</p>"`;

exports[`renders bullet list 1`] = `
"<ul>
<li>item one</li>
<li>item two
<ul>
<li>nested item</li>
</ul>
</li>
</ul>"
`;

exports[`renders checkbox list 1`] = `
"<ul>
<li class=\\"checkbox-list-item\\"><span class=\\"checkbox \\">[ ]</span>unchecked</li>
<li class=\\"checkbox-list-item\\"><span class=\\"checkbox checked\\">[x]</span>checked</li>
</ul>"
`;

exports[`renders code block 1`] = `
"<pre><code>this is indented code
</code></pre>"
`;

exports[`renders code fence 1`] = `
"<pre><code class=\\"language-javascript\\">this is code
</code></pre>"
`;

exports[`renders code marks 1`] = `"<p>this is <code>inline code</code> text</p>"`;

exports[`renders color 1`] = `"<p><span style=\\"color: rgb(240, 107, 5)\\"><span style=\\"background-color: rgba(163, 67, 31, 0.2)\\">测试</span></span><span style=\\"color: rgb(136, 49, 204)\\"><span style=\\"background-color: rgba(74, 82, 199, 0.2)\\">文字</span></span><span style=\\"color: rgb(233, 30, 44)\\"><span style=\\"background-color: rgba(200, 21, 182, 0.2)\\">背景</span></span><span style=\\"color: rgb(3, 135, 102)\\"><span style=\\"background-color: rgba(3, 135, 102, 0.2)\\">颜色</span></span></p>"`;

exports[`renders headings 1`] = `
"<h1>Heading 1</h1>
<h2>Heading 2</h2>
<h3>Heading 3</h3>
<h4>Heading 4</h4>"
`;

exports[`renders highlight marks 1`] = `"<p>this is <span class=\\"highlight\\">highlighted</span> text</p>"`;

exports[`renders horizontal rule 1`] = `"<hr>"`;

exports[`renders image 1`] = `"<p><img src=\\"https://lorempixel.com/200/200\\" alt=\\"caption\\"></p>"`;

exports[`renders image with alignment 1`] = `"<p><img src=\\"https://lorempixel.com/200/200\\" alt=\\"caption\\" title=\\"left-40\\"></p>"`;

exports[`renders info notice 1`] = `
"<div class=\\"notice notice-info\\">
<p>content of notice</p>
</div>"
`;

exports[`renders italic marks 1`] = `"<p>this is <em>italic</em> text</p>"`;

exports[`renders italic marks 2`] = `"<p>this is <em>also italic</em> text</p>"`;

exports[`renders link marks 1`] = `"<p>this is <a href=\\"https://www.example.com\\">linked</a> text</p>"`;

exports[`renders math 1`] = `
"<p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\" display=\\"block\\"><semantics><mrow><mi>x</mi><mo>=</mo><mfrac><mrow><mo>−</mo><mi>b</mi><mo>±</mo><msqrt><mrow><msup><mi>b</mi><mn>2</mn></msup><mo>−</mo><mn>4</mn><mi>a</mi><mi>c</mi></mrow></msqrt></mrow><mrow><mn>2</mn><mi>a</mi></mrow></mfrac></mrow><annotation encoding=\\"application/x-tex\\">x = {-b \\\\pm \\\\sqrt{b^2-4ac} \\\\over 2a}</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.4306em;\\"></span><span class=\\"mord mathnormal\\">x</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:2.2764em;vertical-align:-0.686em;\\"></span><span class=\\"mord\\"><span class=\\"mord\\"><span class=\\"mopen nulldelimiter\\"></span><span class=\\"mfrac\\"><span class=\\"vlist-t vlist-t2\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:1.5904em;\\"><span style=\\"top:-2.314em;\\"><span class=\\"pstrut\\" style=\\"height:3em;\\"></span><span class=\\"mord\\"><span class=\\"mord\\">2</span><span class=\\"mord mathnormal\\">a</span></span></span><span style=\\"top:-3.23em;\\"><span class=\\"pstrut\\" style=\\"height:3em;\\"></span><span class=\\"frac-line\\" style=\\"border-bottom-width:0.04em;\\"></span></span><span style=\\"top:-3.677em;\\"><span class=\\"pstrut\\" style=\\"height:3em;\\"></span><span class=\\"mord\\"><span class=\\"mord\\">−</span><span class=\\"mord mathnormal\\">b</span><span class=\\"mspace\\" style=\\"margin-right:0.2222em;\\"></span><span class=\\"mbin\\">±</span><span class=\\"mspace\\" style=\\"margin-right:0.2222em;\\"></span><span class=\\"mord sqrt\\"><span class=\\"vlist-t vlist-t2\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.9134em;\\"><span class=\\"svg-align\\" style=\\"top:-3em;\\"><span class=\\"pstrut\\" style=\\"height:3em;\\"></span><span class=\\"mord\\" style=\\"padding-left:0.833em;\\"><span class=\\"mord\\"><span class=\\"mord mathnormal\\">b</span><span class=\\"msupsub\\"><span class=\\"vlist-t\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.7401em;\\"><span style=\\"top:-2.989em;margin-right:0.05em;\\"><span class=\\"pstrut\\" style=\\"height:2.7em;\\"></span><span class=\\"sizing reset-size6 size3 mtight\\"><span class=\\"mord mtight\\">2</span></span></span></span></span></span></span></span><span class=\\"mspace\\" style=\\"margin-right:0.2222em;\\"></span><span class=\\"mbin\\">−</span><span class=\\"mspace\\" style=\\"margin-right:0.2222em;\\"></span><span class=\\"mord\\">4</span><span class=\\"mord mathnormal\\">a</span><span class=\\"mord mathnormal\\">c</span></span></span><span style=\\"top:-2.8734em;\\"><span class=\\"pstrut\\" style=\\"height:3em;\\"></span><span class=\\"hide-tail\\" style=\\"min-width:0.853em;height:1.08em;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width='400em' height='1.08em' viewBox='0 0 400000 1080' preserveAspectRatio='xMinYMin slice'><path d='M95,702
c-2.7,0,-7.17,-2.7,-13.5,-8c-5.8,-5.3,-9.5,-10,-9.5,-14
c0,-2,0.3,-3.3,1,-4c1.3,-2.7,23.83,-20.7,67.5,-54
c44.2,-33.3,65.8,-50.3,66.5,-51c1.3,-1.3,3,-2,5,-2c4.7,0,8.7,3.3,12,10
s173,378,173,378c0.7,0,35.3,-71,104,-213c68.7,-142,137.5,-285,206.5,-429
c69,-144,104.5,-217.7,106.5,-221
l0 -0
c5.3,-9.3,12,-14,20,-14
H400000v40H845.2724
s-225.272,467,-225.272,467s-235,486,-235,486c-2.7,4.7,-9,7,-19,7
c-6,0,-10,-1,-12,-3s-194,-422,-194,-422s-65,47,-65,47z
M834 80h400000v40h-400000z'/></svg></span></span></span><span class=\\"vlist-s\\">​</span></span><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.1266em;\\"><span></span></span></span></span></span></span></span></span><span class=\\"vlist-s\\">​</span></span><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.686em;\\"><span></span></span></span></span></span><span class=\\"mclose nulldelimiter\\"></span></span></span></span></span></span></span></p>"
`;

exports[`renders math inline 1`] = `"<p><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><msup><mi>a</mi><mn>2</mn></msup><mo>+</mo><msup><mi>b</mi><mn>2</mn></msup><mo>=</mo><msup><mi>c</mi><mn>2</mn></msup></mrow><annotation encoding=\\"application/x-tex\\">a^2+b^2=c^2</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.8974em;vertical-align:-0.0833em;\\"></span><span class=\\"mord\\"><span class=\\"mord mathnormal\\">a</span><span class=\\"msupsub\\"><span class=\\"vlist-t\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.8141em;\\"><span style=\\"top:-3.063em;margin-right:0.05em;\\"><span class=\\"pstrut\\" style=\\"height:2.7em;\\"></span><span class=\\"sizing reset-size6 size3 mtight\\"><span class=\\"mord mtight\\">2</span></span></span></span></span></span></span></span><span class=\\"mspace\\" style=\\"margin-right:0.2222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.8141em;\\"></span><span class=\\"mord\\"><span class=\\"mord mathnormal\\">b</span><span class=\\"msupsub\\"><span class=\\"vlist-t\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.8141em;\\"><span style=\\"top:-3.063em;margin-right:0.05em;\\"><span class=\\"pstrut\\" style=\\"height:2.7em;\\"></span><span class=\\"sizing reset-size6 size3 mtight\\"><span class=\\"mord mtight\\">2</span></span></span></span></span></span></span></span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.8141em;\\"></span><span class=\\"mord\\"><span class=\\"mord mathnormal\\">c</span><span class=\\"msupsub\\"><span class=\\"vlist-t\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.8141em;\\"><span style=\\"top:-3.063em;margin-right:0.05em;\\"><span class=\\"pstrut\\" style=\\"height:2.7em;\\"></span><span class=\\"sizing reset-size6 size3 mtight\\"><span class=\\"mord mtight\\">2</span></span></span></span></span></span></span></span></span></span></span></p>"`;

exports[`renders ordered list 1`] = `
"<ol>
<li>item one</li>
<li>item two</li>
</ol>"
`;

exports[`renders ordered list 2`] = `
"<ol>
<li>item one</li>
<li>item two</li>
</ol>"
`;

exports[`renders plain text as paragraph 1`] = `"<p>plain text</p>"`;

exports[`renders table 1`] = `
"<table>
<tr>
<th>
<p>heading</p></th>
<th style=\\"text-align:center\\">
<p>centered</p></th>
<th style=\\"text-align:right\\">
<p>right aligned</p></th>
</tr>
<tr>
<td>
<p></p></td>
<td style=\\"text-align:center\\">
<p>center</p></td>
<td style=\\"text-align:right\\">
<p></p></td>
</tr>
<tr>
<td>
<p></p></td>
<td style=\\"text-align:center\\">
<p></p></td>
<td style=\\"text-align:right\\">
<p>bottom r</p></td>
</tr>
</table>"
`;

exports[`renders template placeholder marks 1`] = `"<p>this is <span class=\\"placeholder\\">a placeholder</span></p>"`;

exports[`renders tip notice 1`] = `
"<div class=\\"notice notice-tip\\">
<p>content of notice</p>
</div>"
`;

exports[`renders underline marks 1`] = `"<p>this is <underline>underlined</underline> text</p>"`;

exports[`renders underline marks 2`] = `"<p>this is <s>strikethrough</s> text</p>"`;

exports[`renders warning notice 1`] = `
"<div class=\\"notice notice-warning\\">
<p>content of notice</p>
</div>"
`;
